File: 0921 Minimum Add to Make Parentheses Valid/0921.cpp
class Solution {
public:
    int minAddToMakeValid(string s) {
        int ans = 0;
        int extra = 0;

        for (char c: s){
            if (c == '('){
                extra++;
            } else {
                if (extra == 0){
                    ans++;
                } else {
                    extra--;
                }
            }
        }

        return ans + extra;
    }
};

File: 0921 Minimum Add to Make Parentheses Valid/0921.py
class Solution:
    def minAddToMakeValid(self, s: str) -> int:
        ans, extra = 0, 0

        for c in s:
            if c == '(':
                extra += 1
            else:
                if extra == 0:
                    ans += 1
                else:
                    extra -= 1
        
        return ans + extra
